### PR TITLE
cli: warn when untested node version

### PR DESCRIPTION
**Pull request by davidjgoss**
### 🤔 What's changed?

Distinguish between supported (error if not met) and tested (warning if not met) Node.js versions.

### ⚡️ What's your motivation? 

See https://github.com/cucumber/cucumber-js/pull/1958#issuecomment-1073451670

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, docuemntation etc. without changing behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -188,6 +188,9 @@
   "engines": {
     "node": "12 || 14 || >=16"
   },
+  "enginesTested": {
```
Comment (davidjgoss) [inline package.json:191]: Not a standard field but I think it's worth keeping the two things together.

File: src/cli/validate_node_engine_version_spec.ts [hunk 2]
```diff
@@ -3,66 +3,69 @@ import * as sinon from 'sinon'
 import { validateNodeEngineVersion } from './validate_node_engine_version'
 
 describe(validateNodeEngineVersion.name, () => {
+  const readPackageJSON = () => ({
```
Comment (davidjgoss) [inline src/cli/validate_node_engine_version_spec.ts:6]: Thoughts on just dropping this and having the test read the package.json as normal? It'd mean updating the test when we change the supported/tested versions but I don't think that's necessarily a bad thing. Also would potentially catch mistakes with updates to the package.json.

**Review by aurelien-reeves — APPROVED** [review]

File: src/cli/validate_node_engine_version_spec.ts (hunk 2 above)
Comment (mattwynne) [reply src/cli/validate_node_engine_version_spec.ts:6]: 🤷 the original idea was to allow us to keep the examples in the tests more focused/simple and perhaps cover edge cases that aren't in the `package.json` today, although it looks like they've evolved to pretty much mirror what's in `package.json` now. I personally prefer having my tests tell the complete story in one file, otherwise you can end up in a "[magic number](https://en.wikipedia.org/wiki/Magic_number_(programming))" situation.

Still, if you want to be able to test what's actually in the `package.json`, maybe that deserves a specific test or two as well?
